node-hub/opencv-plot/main.py

```diff
@@ -0,0 +1,164 @@
+        0.5,
+        (255, 255, 255),
+        1,
+        1,
+    )
+
+    if plot.width is not None and plot.height is not None:
+        plot.frame = cv2.resize(plot.frame, (plot.width, plot.height))
+
+    if not ci_enabled:
+        if len(plot.frame.shape) >= 3:
+            cv2.imshow("Dora Node: opencv-plot", plot.frame)
+
+            if cv2.waitKey(1) & 0xFF == ord('q'):
+                return True
+
+    return False
+
+
+def main():
+    # Handle dynamic nodes, ask for the name of the node in the dataflow, and the same values as the ENV variables.
+    parser = argparse.ArgumentParser(
+        description="OpenCV Plotter: This node is used to plot text and bounding boxes on an image.")
+
+    parser.add_argument("--name", type=str, required=False, help="The name of the node in the dataflow.",
+                        default="opencv-plot")
+    parser.add_argument("--plot-width", type=int, required=False, help="The width of the plot.", default=None)
+    parser.add_argument("--plot-height", type=int, required=False, help="The height of the plot.", default=None)
+
+    args = parser.parse_args()
+
+    plot_width = os.getenv("PLOT_WIDTH", args.plot_width)
+    plot_height = os.getenv("PLOT_HEIGHT", args.plot_height)
+
+    if plot_width is not None:
+        if isinstance(plot_width, str) and plot_width.isnumeric():
+            plot_width = int(plot_width)
+
+    if plot_height is not None:
+        if isinstance(plot_height, str) and plot_height.isnumeric():
+            plot_height = int(plot_height)
+
+    # check if the code is running in a CI environment (e.g. GitHub Actions) (parse to bool)
+    ci_enabled = os.getenv("CI", False)
+    if ci_enabled == "true":
+        ci_enabled = True
+
+    node = Node(args.name)  # provide the name to connect to the dataflow if dynamic node
+    plot = Plot()
+
+    plot.width = plot_width
+    plot.height = plot_height
+
+    pa.array([])  # initialize pyarrow array
+
+    for event in node:
+        event_type = event["type"]
+
+        if event_type == "INPUT":
+            event_id = event["id"]
+
+            if event_id == "tick":
+                if ci_enabled:
+                    break
+
+                node.send_output(
+                    "tick",
+                    pa.array([]),
+                    event["metadata"]
+                )
+
+            elif event_id == "image":
+                arrow_image = event["value"][0]
+                image = {
+                    "width": np.uint32(arrow_image["width"].as_py()),
+                    "height": np.uint32(arrow_image["height"].as_py()),
+                    "channels": np.uint8(arrow_image["channels"].as_py()),
+                    "data": arrow_image["data"].values.to_numpy().astype(np.uint8)
+                }
+
+                plot.frame = np.reshape(image["data"], (image["height"], image["width"], image["channels"]))
+
+                if plot_frame(plot, ci_enabled):
+                    break
+
```

Comment:
For clarity, can we rename this:
```suggestion
                if not RUNNER_CI:
                    plot_frame(plot)
                     if cv2.waitKey(1) & 0xFF == ord('q'):
                         break
```